src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java

```diff
@@ -988,6 +1013,9 @@ public ArgumentDefinition(
             //null collections have been initialized by createCollection which is called in handleArgumentAnnotation
             //this is optional if it's specified as being optional or if there is a default value specified
             this.optional = annotation.optional() || ! this.defaultValue.equals(NULL_STRING);
+
+            this.maxValue = annotation.maxValue();
+            this.minValue = annotation.minValue();
```

Comment:
I check both on `ArgumentDefinition`construction, for both hard and soft boundaries.